src/services/database/sqlite/sqlite_database_logs.cpp

```diff
@@ -1,20 +1,26 @@
-#include <fmt/format.h>
 #include "sqlite_database.h"
+#include <fmt/format.h>
 
-std::vector<record_log_t> SqliteDatabase::getRecentLogs(int system_id, int limit) {
-    const std::lock_guard<std::mutex> lock(databaseMutex); // Acquire database lock for this scope
+std::vector<record_log_t> SqliteDatabase::getRecentLogs(int system_id, int limit, int beforeId) {
+    const std::lock_guard<std::mutex> lock(databaseMutex);  // Acquire database lock for this scope
 
     std::vector<record_log_t> logs;
     try {
-        SQLite::Statement query(*db, fmt::format("select * from logs where system_id={} order by timestamp asc limit {}", system_id, limit));
+        std::string whereConditions = fmt::format("system_id={}", system_id);
+        if (beforeId != -1) {
+            whereConditions += fmt::format(" and id < {}", beforeId);
```

Comment:
Had to think about this for a second but the `where` clause (vs `offset`) is the best way to do it for large data sets, so this looks good.

Reply:
On second thought, see above comment about switching to using `limit` and `offset` so we can order by whatever field we want and can specify with start index. This means a select query will be grabbing and filtering from _all_ rows, but we should test -- The performance hit might not be that bad.